src/main/python/tensorframes/core.py

```diff
@@ -377,6 +378,23 @@ def analyze(dframe):
     """
     return DataFrame(_java_api().analyze(dframe._jdf), _sql)
 
+def append_shape(dframe, col, size):
+    """Append extra metadata for a dataframe that
+     describes the numerical shape of the content.
+
+     This method is useful when a dataframe contains non-scalar tensors, for which the shape must be checked beforehand.
+
+     Note: nullable fields are not accepted.
+
+     The function [print_schema] lets users introspect the information added to the DataFrame.
+
+    :param dframe: a Spark DataFrame
+    :param col: a Column expression
+    :param size: a shape corresponding to the tensor
```

Comment:
You can link to the documentation in python: https://www.tensorflow.org/programmers_guide/tensors#shape
This is important for people to understand the order of the elements.